src/charm.py

```diff
@@ -213,18 +248,16 @@ def _amf_pebble_layer(self) -> Layer:
             }
         )
 
-    @property
-    def _default_database_info(self) -> dict:
+    def _get_default_database_info(self) -> dict:
         """Returns the database data.
 
         Returns:
             Dict: The database data.
         """
-        if not self._default_database_is_available:
-            None
+        if not self._default_database_is_available():
+            return {}
```

Comment:
Are we handling the empty relation data at the time of populating the config file? If we accept empty relation as "valid" we'll be populating the config file wrongly, ending up with the charm unable to connect to the db for an unknown reason. Have we tested what's the behavior of the charm if this happens? Does the start command complain?  Does that make sense?

Reply:
I don't think we'll populate the config wrongly, because `database_url=self._get_default_database_info()["uris"].split(",")[0]` will fail when it gets an empty dict. 
In any case, I think it's better to raise an error here. 
WDYT?

Reply:
I don't really see how that can happen, we are checking for database to be available and for the db info to be available too in this very same handler, if those are not available we are not rendering the config, we are in `waiting` status instead.
And if what you are saying happens, then the config won't be populated with an empty `db url` the run will fail with a `key error` instead.
But then how about the `nrf-url` how do we double check for that if this was true?

Reply:
Error added to be inline with the rest of the charms.

Reply:
I haven't thought of this @Gmerold, great point.

Reply:
About the nrf URL, we check at the library level whatever we read must be a URL, whatever comes straight out of it, should be properly already properly checked. And this check
```python
if not self._nrf_requires.nrf_url
```
Should also prevent an empty string, doesn't it?

Reply:
Feel free to resolve as well.

Reply:
Thanks for the clarification on the `nrf-url` :) @sanchezfdezjavier